force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -101,17 +115,20 @@ public with sharing class ClassesObjectsMethods {
      * @return The number of vowels in the string.
      */
     public static Integer countVowels(String str) {
-        // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        Integer vowelCount = 0;
+        String testString = '';
+        Integer beginIndex = 0;
+        Integer endIndex = 1;    
+        for (Integer i = 0; i < str.length(); i++) {
+            testString = str.substring(beginIndex, endIndex);
+            if (testString == 'a' || testString == 'e' || testString == 'i' || testString == 'o' || testString == 'u') {
```

Comment:
You can probably get this done easier with a Set<String> of vowel chars, or using the contains() or containsAny methods. for Set/String classes. Look them and see how you could apply them to simplify your code. Doing it low level like this is impressive, though.